src/clj/fluree/db/json_ld/policy/rules.cljc

```diff
@@ -84,68 +72,106 @@
      policy-map
      (:on-property restriction-map))))
 
-(defn restriction-map
-  [restriction]
-  (let [id          (util/get-id restriction) ;; @id name of restriction
-        on-property (when-let [props (util/get-all-ids restriction const/iri-onProperty)]
-                      (set props)) ;; can be multiple properties
-        on-class    (when-let [classes (util/get-all-ids restriction const/iri-onClass)]
-                      (set classes))
-        src-query   (util/get-first-value restriction const/iri-query)
-        query       (if (map? src-query)
+(defn parse-targets
+  [db target-exprs]
+  (let [out-ch (async/chan 2)
+        in-ch  (async/to-chan! (mapv #(or (util/get-id %) (util/get-value %)) target-exprs))]
+    (async/pipeline-async 2
+                          out-ch
+                          (fn [target-expr ch]
+                            (if (map? target-expr)
+                              ;; maps are query where clauses
+                              (-> (dbproto/-query db (assoc target-expr "select" "?$target"))
```

Comment:
Is there a reason to require special variable syntax? I'd favor people writing a "normal" query with their own select syntax... but should do some validation that "select" is bound to just a single var.

Certainly a judgement call... so not opposed to this approach, just on surface I don't prefer it.

Reply:
I did the hard-coded `?$target` variable in the spirit of our current harcoded vars: `?$identity` and `?$this`. It simplifies parsing, since we don't have to validate select and we can just document what we expect, but I don't think it's a big deal to just require them to return a single var anyways.

Reply:
In the "virtual-graphs" (full text search, vector search) we allow people to format the queries using their own vars... but just when parsing the query, verify the `:select` is in a suitable format else provide an error with useful feedback.

The reason ?$this, ?$identity exists is because we inject those with `:values` into the query so we need a fixed variable name. I don't think that is happening here since the query is run once initially.